lib/execute.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Execute do
+  def run_module(module) do
+    run_until_failure(module, &run_koan/2)
+  end
+
+  defp run_until_failure(module, callback) do
+    Enum.reduce_while(module.all_koans, :passed, fn(it, _) ->
+      module
+      |> callback.(it)
+      |> continue?
+    end)
+  end
+
+  defp continue?(:passed), do: {:cont, :passed}
+  defp continue?(result), do: {:halt, result}
+
+  def run_koan(module, name, args \\ []) do
+    parent = self()
+    spawn fn -> exec(module, name, args, parent) end
+    receive do
+      :ok   -> :passed
+      error -> {:failed, error, module, name}
+    end
+  end
+
+  defp exec(module, name, args, parent) do
+    result = apply(module, name, args)
+    send parent, expand(result)
+    Process.exit(self(), :kill)
```

Comment:
Leaving it out fails the tests. My best guess is that it affects how children get cleaned up, if tall.